File: src/main/java/handler/FanduelHandler.java
package handler;

import api.ApiCaller;
import api.DataCollector;
import collect.dfs.Fanduel;
import util.AWSClient;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class FanduelHandler {
    private DataCollector dataCollector = new DataCollector(new ApiCaller());
    private Fanduel fanduel = new Fanduel(dataCollector);
    private AWSClient AWSClient = new AWSClient();

    public List<Map<String, Object>> handleRequest(Map<String, String> input) {
        String invocationType = input.getOrDefault("invocationType", "web");
        String date = input.get("date");
        List<Map<String, Object>> result = fanduel.getAllContestData(date);
        if (invocationType.equals("pipeline")) {
            AWSClient.uploadToS3("fanduelData.json", result);
            return new ArrayList<>();
        } else
            return result;
    }
}

File: src/main/kotlin/handler/FanduelHandler.kt
package handler

import collect.dfs.FanduelContests
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import util.AWS

class FanduelHandler {
    suspend fun handleRequest(input: Map<String, String>): List<Map<String, Any>> {
        val date = input.getValue("date")
        val invocationType = input.getOrDefault("invocationType", "web")
        val fanduelData = getFanduelContestData(date)
        return if (invocationType == "pipeline") {
            uploadToS3("fanduelData.json", fanduelData)
            listOf()
        } else fanduelData
    }

    fun getFanduelContestData(date: String): List<Map<String, Any>> {
        return FanduelContests().getFanduelContestData(date)
    }

    suspend fun uploadToS3(fileName: String, objectToUpload: Any) {
        return withContext(Dispatchers.Default) { AWS().uploadToS3(fileName, objectToUpload) }
    }
}